### PR TITLE
Added Hardware interface VAAPI and QSV

### PR DESCRIPTION
**Pull request by ryanwclark1**
LibVA - implementation for VAAPI
LibVPL - Intel Video Processing Library
Updated Readme with source library link

[FFMPEG wiki on VAAPI](https://trac.ffmpeg.org/wiki/Hardware/VAAPI)
[FFMPEG wiki on Quicksync/QSV](https://trac.ffmpeg.org/wiki/Hardware/QuickSync)

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1027,6 +1028,51 @@ RUN \
...
+    -Ddisable_drm=false \
+    -Dwith_x11=no \
+    -Dwith_glx=no \
+    -Dwith_wayland=no \
+    -Dwith_win32=no \
+    -Dwith_legacy=[] \
+    -Denable_docs=false && \
+  ninja -j$(nproc) -vC build install
+
+# bump: libvpl /LIBVPL_VERSION=([\d.]+)/ https://github.com/intel/libvpl.git|^0
```
Comment (wader) [inline Dockerfile:1054]: Change to `...|^2` and it should work

File: Dockerfile [hunk 2]
```diff
@@ -1027,6 +1028,51 @@ RUN \
   sed -i 's/-ljxl_cms/-ljxl_cms -lstdc++ /' /usr/local/lib/pkgconfig/libjxl_cms.pc && \
   sed -i 's/-ljxl_threads/-ljxl_threads -lstdc++ /' /usr/local/lib/pkgconfig/libjxl_threads.pc
 
+# requires libdrm
+# bump: libva /LIBVA_VERSION=([\d.]+)/ https://github.com/intel/libva.git|^0
```
Comment (wader) [inline Dockerfile:1032]: Change to `...|^2` and it should work

Comment (wader) [reply Dockerfile:1032]: Usually can use `*` but there seems to lots of random tags

File: Dockerfile [hunk 3]
```diff
@@ -1027,6 +1028,51 @@ RUN \
   sed -i 's/-ljxl_cms/-ljxl_cms -lstdc++ /' /usr/local/lib/pkgconfig/libjxl_cms.pc && \
   sed -i 's/-ljxl_threads/-ljxl_threads -lstdc++ /' /usr/local/lib/pkgconfig/libjxl_threads.pc
 
+# requires libdrm
```
Comment (wader) [inline Dockerfile:1031]: Don't see any .a files for libdrm but maybe libva only needs the headers?

**Comment by wader** [comment]
Thanks! looks good, only have some minor comments. I don't the hardware to test this, but i assume you have tried? :)

Does this usually require volume mount in some `/dev` device into the container to work? something to add to the README?

**Comment by ryanwclark1** [comment]
> Thanks! looks good, only have some minor comments. I don't the hardware to test this, but i assume you have tried? :)
> 
> Does this usually require volume mount in some `/dev` device into the container to work? something to add to the README?

@wander,
I tested using the Intel ARC 310 and 380.  I did not try with an AMD GPU which can use the VAAPI interface, I do have one and will report back on this.  NVIDIA, as I'm sure you know, is separate and distinct.

My tests were done using a separate base image with some additional user space tools/drivers and copying in the ffmpeg binary, I used --device /dev/dri in the docker run command to pass my gpu into the container.  I will test using the "Run directly" and the "Shell Alias" methods described in the README.  Will report back.

**Comment by wader** [comment]
> > Thanks! looks good, only have some minor comments. I don't the hardware to test this, but i assume you have tried? :)
> > Does this usually require volume mount in some `/dev` device into the container to work? something to add to the README?
> 
> @wander, I tested using the Intel ARC 310 and 380. I did not try with an AMD GPU which can use the VAAPI interface, I do have one and will report back on this. NVIDIA, as I'm sure you know, is separate and distinct.

👍

There is draft PR https://github.com/wader/static-ffmpeg/pull/482 about nvidia but it seems to be a mess to get working with  musl, and i guess it might involve linking with some random binary from nvidia? not great if so.

> My tests were done using a separate base image with some additional user space tools/drivers and copying in the ffmpeg binary, I used --device /dev/dri in the docker run command to pass my gpu into the container. I will test using the "Run directly" and the "Shell Alias" methods described in the README. Will report back.

Great! just some short example how to use it would be good and enough i think.

**Comment by wader** [comment]
Force pushed fixed for ^2 and conflict with master

**Comment by wader** [comment]
Thanks! let's merge this to not end up with conflicts all the time. Feel free to add docs

**Pull request merged**